### PR TITLE
Docker images

### PR DESCRIPTION
**Pull request by philomath213**
## Description
Update GitHub Actions Workflow to build and push docker images on release for multiple python versions (3.6, 3.7 and 3.8).

#### EDIT:
Tags list:
Nightly Images (build from master branch):
- latest-py36
- latest-py37
- latest-py38

Release Images (build on release):
- \<version\>-py36
- \<version\>-py37
- latest, \<version\>, \<version\>-py38

Latest release image for Python 3.8 is considered the latest stable release.

## Type of change

- [ ] Added/Modified tutorials/docs
- [ ] Refactoring
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

* [ ] I have added tests for my changes
* [ ] I did follow the [contribution guidelines](#)

### CODE REVIEW

File: .github/workflows/docker-image.yml [hunk 1]
```diff
@@ -1,23 +1,38 @@
 name: Docker Image
 
 on:
-  push:
-    branches:
-      - master
+  release:
+    types: [created]
```
Comment (youben11) [inline .github/workflows/docker-image.yml:5]: We will need to have both on release and on push here, as users may want to use the nightly version as well

File: .github/workflows/docker-image.yml [hunk 2]
```diff
@@ -1,23 +1,38 @@
...
+      - name: build image
+        run: docker build -t $IMAGE -f docker-images/Dockerfile-py${{ matrix.python-version }} .
+
+      - name: docker login
+        run: echo "${{ secrets.DOCKER_PASSWORD }}" |  docker login -u ${{ secrets.DOCKER_LOGIN }} --password-stdin
+
+      - name: tag and push image
+        run: |
+          docker tag $IMAGE $IMAGE:${{ github.event.release.tag_name }}-py${{ matrix.python-version }}
+          docker push $IMAGE:${{ github.event.release.tag_name }}-py${{ matrix.python-version }} .
```
Comment (youben11) [inline .github/workflows/docker-image.yml:32]: Tagging here will depends on whether it's a release or not

File: .github/workflows/docker-image.yml (hunk 1 above)
Comment (philomath213) [reply .github/workflows/docker-image.yml:5]: the nightly version should be only for python 3.8, or for all python versions?

Comment (youben11) [reply .github/workflows/docker-image.yml:5]: Why not all?

File: .github/workflows/docker-image.yml (hunk 2 above)
Comment (philomath213) [reply .github/workflows/docker-image.yml]: Fixed.

File: .github/workflows/docker-image.yml [hunk 3]
```diff
@@ -4,20 +4,46 @@ on:
...
-          add_git_labels: true
+
+      - name: build image
+        run: docker build -t $IMAGE -f docker-images/Dockerfile-py${{ matrix.python-version }} .
+
+      - name: docker login
+        run: echo "${{ secrets.DOCKER_PASSWORD }}" | docker login -u ${{ secrets.DOCKER_LOGIN }} --password-stdin
+
+      - name: tag and push latest nightly images
+        if: github.event_name == 'push' && github.ref == 'refs/heads/master'
```
Comment (youben11) [inline .github/workflows/docker-image.yml:33]: The push event is only on master, why do we need to check github.ref?

File: .github/workflows/docker-image.yml [hunk 4]
```diff
@@ -4,20 +4,46 @@ on:
...
+        run: |
+          docker tag $IMAGE $IMAGE:latest-py${{ matrix.python-version }}
+          docker push $IMAGE:latest-py${{ matrix.python-version }}
+
+      - name: tag and push release image
+        if: github.event_name == 'release' && github.event.action == 'created'
```
Comment (youben11) [inline .github/workflows/docker-image.yml:39]: same for github.event.action

File: .github/workflows/docker-image.yml (hunk 3 above)
Comment (philomath213) [reply .github/workflows/docker-image.yml]: double checking, sorry I forget that.

**Pull request merged**